### DISCUSSION TITLE
Any solution for using flagd configuration file for Java Memory Provider

### DISCUSSION THREAD

**Discussion by cartermfv** [open-feature/community#266] [Q&A] [answered]
Hi, I want using flagd configuration [file](https://github.com/open-feature/flagd/blob/main/config/samples/example_flags.json)
for Java Memory provider [flag](https://github.com/open-feature/java-sdk/blob/main/src/main/java/dev/openfeature/sdk/providers/memory/Flag.java) Object
Any library for convert flagd configuration file to Java Memory Flag class ?
Thank you very much

**Comment by Kavindu-Dodan** [comment, marked as answer]
@cartermfv flagd Java provider implementation now has the support for an offline mode. This was released with the version [v0.6.6](https://github.com/open-feature/java-sdk-contrib/releases/tag/dev.openfeature.contrib.providers.flagd-v0.6.6) and you can find the updated documentation [here](https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/README.md#in-process-resolver).

For example, if you would like to use [example_flags.json](https://github.com/open-feature/flagd/blob/main/config/samples/example_flags.json) as your feature flags, you can initilize flagd provider as below,

```java
FlagdProvider flagdProvider = new FlagdProvider(
        FlagdOptions.builder()
                .resolverType(Config.Evaluator.IN_PROCESS)
                .offlineFlagSourcePath("example_flags.json")
                .build());
```

This mode was added to improve the local developer experience and help with testing. However, it is totally acceptable to use with a self-contained application.

Let me know if this helps you.

**Reply by cartermfv** [reply]
Thank you very much
